src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -49,6 +49,7 @@ internal partial class PackageRestoreDataSource : ChainedProjectValueDataSourceB
         // |_________________________________________|    |_________________________________________|
         //
 
+        private readonly NuGetRestoreCycleDetector _cycleDetector = new(5);
```

Comment:
Do we have a sense of how long it would take to restore five times? If it's not very long, I'd vote to increase this threshold.

Reply:
You can maybe use https://docs.microsoft.com/en-us/nuget/visual-studio-extensibility/nuget-api-in-visual-studio#ivsnugetprojectupdateevents-interface to detect the timing of restore start/end. 

We can also add other events if they'd make life easier for you to do detect these loops.

Reply:
In the next commit the threshold will be increased to 10